catalog/transformer/catalog-transformer-generic-impl/src/main/java/ddf/catalog/transformer/generic/xml/impl/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.generic.xml.impl;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.transformer.generic.xml.SaxEventHandler;
+
+public class SaxEventHandlerDelegate extends DefaultHandler {
+
+    private static SAXParser parser;
+
+    private List<SaxEventHandler> eventHandlers;
+
+    private String namespace;
+
+    private InputStream stream;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SaxEventHandlerDelegate.class);
+
+    public SaxEventHandlerDelegate() {
+        try {
+            // Read set up
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+
+            factory.setSchema(null);
+            factory.setNamespaceAware(true);
+            factory.setValidating(false);
+            //            factory.setFeature("http://xml.org/sax/features/namespaces", false);
+            factory.setFeature("http://xml.org/sax/features/validation", false);
+            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar",
+                    false);
+            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd",
+                    false);
+
+            parser = factory.newSAXParser();
```

Comment:
It looks like if an exception is thrown here, parser ends up being null, which means this class would only ever return null pointers if this case were encountered. Instead of catching and continuing, it'd probably make more sense to simply throw the error all the way up.